GitUI/Editor/FileViewer.cs

```diff
@@ -415,34 +471,42 @@ public void ViewCurrentChanges(GitItemStatus item, bool isStaged, [CanBeNull] Ac
         {
             // BUG why do we call getStatusAsync() twice
 
-            if (!isSubmodule)
-            {
-                _async.LoadAsync(
-                    () => (patch: Module.GetCurrentChanges(fileName, oldFileName, staged, GetExtraDiffArguments(), Encoding), openWithDifftool),
-                    patch => ViewStagingPatch(patch.patch, patch.openWithDifftool));
-            }
-            else if (getStatusAsync() != null)
-            {
-                ViewPatchAsync(() =>
+            ShowOrDeferAsync(
+                fileName,
+                () =>
+                {
+                    if (!isSubmodule)
                     {
-                        var status = ThreadHelper.JoinableTaskFactory.Run(() => getStatusAsync());
-                        if (status == null)
-                        {
-                            return (text: $"Submodule \"{fileName}\" has unresolved conflicts",
-                                    openWithDifftool: null /* not applicable */);
-                        }
+                        return _async.LoadAsync(
+                            () => (patch: Module.GetCurrentChanges(fileName, oldFileName, staged, GetExtraDiffArguments(), Encoding), openWithDifftool),
+                            patch => ViewStagingPatch(patch.patch, patch.openWithDifftool));
+                    }
+                    else if (getStatusAsync() != null)
+                    {
+                        return ViewPatchAsync(
+                            () =>
+                            {
+                                var status = ThreadHelper.JoinableTaskFactory.Run(() => getStatusAsync());
+                                if (status == null)
+                                {
+                                    return (text: $"Submodule \"{fileName}\" has unresolved conflicts",
```

Comment:
I understand this is the existing code, but it is something we'll need to convert to a `TranslationString` as well.